localstack-core/localstack/services/apigateway/next_gen/execute_api/integrations/http.py

```diff
@@ -1,20 +1,137 @@
+    cookies: Optional[dict[str, str]]
+    timeout: Optional[int]
+    allow_redirects: Optional[bool]
+    stream: Optional[bool]
+    verify: Optional[bool]
+    # TODO: check if there was a situation where we'd pass certs?
+    cert: Optional[str | tuple[str, str]]
+
 
-class RestApiHttpIntegration(RestApiIntegration):
+class BaseRestApiHttpIntegration(RestApiIntegration):
+    @staticmethod
+    def _get_default_headers(context: RestApiInvocationContext) -> dict[str, str]:
+        # passing full context as the trace-id will probably also come from it
+        # TODO: get right casing in case we can override them with mappings/override?
+        return {
+            "x-amzn-apigateway-api-id": context.api_id,
+            "x-amzn-trace-id": "",  # TODO
+            "user-agent": f"AmazonAPIGateway_{context.api_id}",
+            "accept-encoding": "gzip,deflate",
+        }
+
+
+class RestApiHttpIntegration(BaseRestApiHttpIntegration):
     """
     This is a REST API integration responsible to send a request to another HTTP API.
     https://docs.aws.amazon.com/apigateway/latest/developerguide/setup-http-integrations.html#api-gateway-set-up-http-proxy-integration-on-proxy-resource
     """
 
     name = "HTTP"
 
+    def invoke(self, context: RestApiInvocationContext) -> Response:
+        integration_req: IntegrationRequest = context.integration_request
+        method = integration_req["http_method"]
+
+        default_apigw_headers = self._get_default_headers(context)
+        default_apigw_headers.update(integration_req["headers"])
+
+        request_parameters: SimpleHttpRequest = {
+            "method": method,
+            "url": integration_req["uri"],
+            "params": integration_req["query_string_parameters"],
+            "headers": default_apigw_headers,
+        }
+
+        if method not in NO_BODY_METHODS:
+            request_parameters["data"] = integration_req["body"]
+
+        # TODO: configurable timeout (29 by default)
+        # request_parameters["timeout"] = 29
+        # TODO: check for redirects
+        # request_parameters["allow_redirects"] = False
 
-class RestApiHttpProxyIntegration(RestApiIntegration):
+        request_response = requests.request(**request_parameters)
+
+        return Response(
+            response=request_response.content,
+            status=request_response.status_code,
+            headers=Headers(dict(request_response.headers)),
+        )
+
+
+class RestApiHttpProxyIntegration(BaseRestApiHttpIntegration):
     """
     This is a simplified REST API integration responsible to send a request to another HTTP API by proxying it almost
     directly.
     https://docs.aws.amazon.com/apigateway/latest/developerguide/setup-http-integrations.html#api-gateway-set-up-http-proxy-integration-on-proxy-resource
     """
 
     name = "HTTP_PROXY"
+
+    def invoke(self, context: RestApiInvocationContext) -> Response:
+        invocation_req: InvocationRequest = context.invocation_request
```

Comment:
I think this might be wrong, we need to use the integration method, with maybe the same fallback mechanism in case of `ANY`. Should create a new test, and extract a helper method we can also use in the Integration Request handler?